pwndbg/commands/spray.py

```diff
@@ -52,7 +59,28 @@ def spray(addr, length, value) -> None:
     else:
         value_bytes = cyclic(length, n=pwndbg.gdblib.arch.ptrsize)
 
-    try:
-        pwndbg.gdblib.memory.write(addr, value_bytes)
-    except gdb.MemoryError as e:
-        print(M.error(e))
+    if only_funcptrs:
+        try:
+            mem = pwndbg.gdblib.memory.read(addr, length)
+        except gdb.MemoryError as e:
+            print(M.error(e))
+            return
+
+        addresses_written = 0
+        ptrsize = pwndbg.gdblib.arch.ptrsize
+        for i in range(0, len(mem) - length % ptrsize, ptrsize):
+            ptr_candidate = struct.unpack(pwndbg.gdblib.arch.fmt, mem[i : i + ptrsize])[0]
+            page = pwndbg.gdblib.vmmap.find(ptr_candidate)
+            if page is not None and page.execute:
+                try:
+                    pwndbg.gdblib.memory.write(addr + i, value_bytes[i : i + ptrsize])
+                    addresses_written += 1
+                except gdb.MemoryError as e:
+                    print(M.error(e))
+                    return
+        print(M.notice(f"Overwritten {addresses_written} function pointers"))
+    else:
+        try:
+            pwndbg.gdblib.memory.write(addr, value_bytes)
+        except gdb.MemoryError as e:
+            print(M.error(e))
```

Comment:
nit: I think we can probably combine the various `gdb.MemoryError` exception handlers into just one or two?

Reply:
Fixed in https://github.com/pwndbg/pwndbg/pull/1819